src/shims/env.rs

```diff
@@ -74,34 +133,43 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let name_ptr = this.read_scalar(name_op)?.not_undef()?;
         let value_ptr = this.read_scalar(value_op)?.not_undef()?;
-        let value = this.read_os_str_from_c_str(value_ptr)?;
+        let value = this.read_os_str_from_target_str(value_ptr)?;
```

Comment:
This is where the "invalid use of NULL ptr" comes from. On Unix, I don't think it is legal to set an env var to NULL. On Windows, that is the way to remove the env var.

So, you'll have to add some target-specific code for that here or in `setenvironmentvariablew`.

Reply:
Ah @bjorn3 already said that, sorry for the redundancy.